### PR TITLE
Added modules to test CIDR blocklisting of Clients

### PR DESCRIPTION
**Pull request by pdhiran**
Automation of test : 
CEPH-83575008 - Verify that CIDR blocklisting of the ceph client IP's is working as expected
https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83575008

Added test to verify blocklisting of Ceph clients via IP and CIDR notation of client Subnet.
Steps:
1. Create two RBD pools.
2. Create images and mount them on all clients.
3. Deployment includes 6 client nodes belonging to 3 different subnets.
4. Write data to images, and check if they have been added to watchers list.
5. Add single client IP to blocklist and verify IO capabilities.
6. Add the CIDR of the client, blocking all IO from all clients from that Subnet.

### CODE REVIEW

**Comment by pdhiran** [comment]
Pass log on regular cluster with 1 client : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-8YOJAH

Tests in progress with 6 clients. Will update results soon.

**Review by SrinivasaBharath — APPROVED** [review]
LGTM

**Review by neha-gangadhar — COMMENTED** [review]
Looks good

**Comment by pdhiran** [comment]
Pass logs with multiple clients in multiple subnets : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-KVGFHX 

Updated conf from 6 clients to 4 clients.

File: tests/rados/test_cidr_blocklisting.py [hunk 1]
```diff
@@ -18,17 +21,279 @@ def run(ceph_cluster, **kw):
...
+        log.info(
+            "completed writing data into all the images on the pools"
+            "Sleeping for 60 seconds for the watchers to be updated..."
+        )
+
+        # Sleeping for 40 seconds and checking if clients are updated as watchers...
+        time.sleep(60)
+
+        for client in client_nodes:
+            log.debug(f"Client selected to write IOs is : {client.hostname}")
```
Comment (neha-gangadhar) [inline tests/rados/test_cidr_blocklisting.py:113]: can you please check the logger here ?

File: ceph/rados/core_workflows.py [hunk 2]
```diff
@@ -2877,3 +2877,218 @@ def set_noautoscale_flag(self, retry: int = 15) -> bool:
 
         log.error("Noautoscale flag not set on the cluster. Returning Fail..")
         return False
+
+    def create_rbd_image(self, pool_name, img_name, **kwargs):
```
Comment (harshkumarRH) [inline ceph/rados/core_workflows.py:2881]: This module is already available in RBD workflows, Is there a need to create a duplicate module in our workflow to achieve the same thing?

Comment (pdhiran) [reply ceph/rados/core_workflows.py:2881]: I have not reviewed the RBD test modules. I feel it's fine. Do you feel that this should be removed?

File: tests/rados/test_cidr_blocklisting.py (hunk 1 above)
Comment (pdhiran) [reply tests/rados/test_cidr_blocklisting.py]: Updated log lines.

File: ceph/rados/core_workflows.py (hunk 2 above)
Comment (neha-gangadhar) [reply ceph/rados/core_workflows.py:2881]: I had the same thought. The basic RBD functions are already exisiting in the repo. Im good with both, keep them or remove them.

Comment (harshkumarRH) [reply ceph/rados/core_workflows.py:2881]: IMO wee can keep them for now and but should definitely review duplicate modules later, maybe once we are done with tier automation

File: conf/quincy/rados/6-node-cluster-4-clients.yaml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+          - provider_net_cci_12
+        role:
+          - mon
+          - mgr
+          - mds
+          - rgw
+      node7:
+        image-name:
+          openstack: RHEL-9.3.0-x86_64-ga-latest
+          ibmc: rhel-91-server-released
```
Comment (harshkumarRH) [inline conf/quincy/rados/6-node-cluster-4-clients.yaml:57]: Why do we need to identify the Distro image?

File: conf/pacific/rados/6-node-cluster-4-clients.yaml [hunk 4]
```diff
@@ -54,46 +54,30 @@ globals:
...
           ibmc: rhel-87-server-released
         networks:
-          - provider_net_cci_9
+          - provider_net_cci_10
         role:
           - client
       node10:
         image-name:
           openstack: RHEL-7.9-x86_64-ga-latest
           ibmc: rhel-server-79-x86-64-kvm
```
Comment (harshkumarRH) [inline conf/pacific/rados/6-node-cluster-4-clients.yaml:79]: If all the other clients are on RHEL8, any reason as to why this is on RHEL7?

Comment (pdhiran) [reply conf/pacific/rados/6-node-cluster-4-clients.yaml:79]: This is on pacific. On quincy and reef, it's RHEL 9.

I can change this as well to RHEL 8 if needed.

File: conf/quincy/rados/6-node-cluster-4-clients.yaml (hunk 3 above)
Comment (pdhiran) [reply conf/quincy/rados/6-node-cluster-4-clients.yaml:57]: To let know which version of client needs to be installed.

File: conf/pacific/rados/6-node-cluster-4-clients.yaml (hunk 4 above)
Comment (pdhiran) [reply conf/pacific/rados/6-node-cluster-4-clients.yaml]: Have Updated to latest RHEL8 image on all clients.

File: conf/quincy/rados/6-node-cluster-4-clients.yaml (hunk 3 above)
Comment (harshkumarRH) [reply conf/quincy/rados/6-node-cluster-4-clients.yaml:57]: Why can it not be same as other nodes?

Comment (pdhiran) [reply conf/quincy/rados/6-node-cluster-4-clients.yaml:57]: I did not understand. what others?

**Review by harshkumarRH — APPROVED** [review]
LGTM

**Review by neha-gangadhar — APPROVED** [review]
lgtm

**Pull request merged**